### PR TITLE
[cmake] Uncomment binaries

### PR DESCRIPTION
**Pull request by IvanKobzarev**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #71246
* __->__ #71157

Differential Revision: [D33528259](https://our.internmc.facebook.com/intern/diff/D33528259)

The root problem: There is no CI check that `BUILD_BINARY=ON` configuration works fine with any combination of build flags.

Uncomment commented by mistake and fix binaries to work with `BUILD_CAFFE2=ON` and `BUILD_CAFFE2=OFF`

1. Move binaries that depend on `BUILD_CAFFE2` into `if(BUILD_CAFFE2)`
2. `torch::jit::Function` not every has graph, using api to convert `toGraphFunction`

Testing:
`BUILD_BINARY=1 BUILD_CAFFE2=1 ./scripts/build_local.sh`

`BUILD_BINARY=1 BUILD_CAFFE2=0 ./scripts/build_local.sh`

### CODE REVIEW

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33528259).

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33528259).

**Comment by seemethere** [comment]
Do we have a CI configuration right now that'll test this code path?

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33528259).

**Review by malfet — APPROVED** [review]
Looks good to me, but please update comment message saying that this change undoing  changes accidentally introduced by https://github.com/pytorch/pytorch/pull/68400

File: binaries/dump_operator_names.cc [hunk 1]
```diff
@@ -29,7 +29,8 @@ void dump_opnames(const Module& m, std::unordered_set<std::string>& opnames) {
   for (const auto& method : methods) {
     const auto& func = method.function();
     std::cout << "function name: " << func.name() << std::endl;
-    torch::jit::Code code(func.graph(), "");
+    auto graph = toGraphFunction(func).graph()->copy();
+    torch::jit::Code code(graph, "");
```
Comment (malfet) [inline binaries/dump_operator_names.cc:33]: Since you are working with stacks anyway, can you perhaps move this change to a separate PR?

**Comment by malfet** [comment]
@seemethere we do not have CI configs for all permutations of options, especially for something like an old caffe2 builds, but we can pass `BUILD_BINARIES=1` to caffe2 build, if those are still relevant

**Comment by IvanKobzarev** [comment]
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33528259).

File: binaries/dump_operator_names.cc (hunk 1 above)
Comment (IvanKobzarev) [reply binaries/dump_operator_names.cc]: @malfet  Moved this change to https://github.com/pytorch/pytorch/pull/71246
